### PR TITLE
Feature/get tests jenkins

### PR DESCRIPTION
**Pull request by JoseSilvaHermida**
- Implemented the get_tests() method for Jenkins (called when using the --tests argument from command line)
- Added unit tests for the Jenkins.get_tests() method

### CODE REVIEW

File: cibyl/sources/jenkins.py [hunk 1]
```diff
@@ -217,6 +221,7 @@ def generate_query_url():
 
             url += f'{api_entrypoint}{query}'
 
+            LOG.debug("URL: %s", url)
```
Comment (cescgina) [inline cibyl/sources/jenkins.py:224]: This is probably a debugging log line that we want to remove. If we want to keep it, then I think we should make a bit more explicit (i.e connecting to jenkins with url...)

File: cibyl/sources/jenkins.py [hunk 2]
```diff
@@ -307,6 +312,75 @@ def get_last_build(self, **kwargs):
 
         return AttributeDictValue("jobs", attr_type=Job, value=job_objects)
 
+    def get_tests(self, **kwargs):
```
Comment (cescgina) [inline cibyl/sources/jenkins.py:315]: The get_tests is missing a speed index, just add `@speed_index({'base': 2})` to decorate the get_tests methods and should be good to go, we can ajust the actual value as more sources implement it and we can compare.

File: cibyl/sources/jenkins.py (hunk 1 above)
Comment (JoseSilvaHermida) [reply cibyl/sources/jenkins.py]: I chose to remove it, because it showed the user personal token in the URL

File: cibyl/sources/jenkins.py (hunk 2 above)
Comment (JoseSilvaHermida) [reply cibyl/sources/jenkins.py:315]: Done, thanks

File: cibyl/sources/jenkins.py [hunk 3]
```diff
@@ -307,6 +311,76 @@ def get_last_build(self, **kwargs):
...
+                    build_id=build['number'],
+                    status=build['result'])
+
+                if build['result'] == 'FAILURE':
+                    LOG.warning("Build %s for job %s failed. No tests to "
+                                "fetch", build['number'], job_name)
+                    continue
+
+                # Get the tests for this build
+                tests_found = self.send_request(
```
Comment (cescgina) [inline cibyl/sources/jenkins.py:365]: This request can return 404 if a job is succesfull and contains no tests. To reproduce run `cibyl --tests --jobs designate`. The simplest solution would be to catch the JenkinsError exception, check if it contains 404 in the error message and warn. If it's a different exception, then raise.

**Review by cescgina — CHANGES_REQUESTED** [review]
Overall looks great, just need to fix the 404 raised by some jobs (see inline comment) and should be good to merge.

**Review by bregman-arie — CHANGES_REQUESTED** [review]
Great job.
Nit: can you also update https://cibyl.readthedocs.io/en/latest/sources.html#arguments-matrix ? Thanks :)

File: cibyl/sources/jenkins.py (hunk 3 above)
Comment (JoseSilvaHermida) [reply cibyl/sources/jenkins.py]: Done. Thanks for the tip

File: cibyl/sources/jenkins.py [hunk 4]
```diff
@@ -356,28 +356,36 @@ def get_tests(self, **kwargs):
...
                 # Get the tests for this build
-                tests_found = self.send_request(
-                    item=f"job/{job_name}/{build['number']}/testReport",
-                    query=self.build_tests_query)
+                try:
+                    tests_found = self.send_request(
+                        item=f"job/{job_name}/{build['number']}/testReport",
+                        query=self.build_tests_query)
+                except JenkinsError as jerr:
+                    if re.search('404', str(jerr)):
```
Comment (adrianfusco) [inline cibyl/sources/jenkins.py:372]: You can check if there is a string in another (you have some examples in different answers here: https://stackoverflow.com/a/3437070/12771230 ). Maybe it's better than using regex for this.

**Review by adrianfusco — APPROVED** [review]
I see that you did all the requested changes. From me +1.

**Review by cescgina — APPROVED** [review]
Everything works now, great job!

File: cibyl/sources/jenkins.py (hunk 4 above)
Comment (JoseSilvaHermida) [reply cibyl/sources/jenkins.py]: That's better, I've just changed that. Thanks a lot!

**Review by bregman-arie — APPROVED** [review]

**Pull request merged**